include/mbgl/util/run_loop.hpp

```diff
@@ -15,21 +15,36 @@
 namespace mbgl {
 namespace util {
 
+typedef void * LOOP_HANDLE;
+
 class RunLoop : private util::noncopyable {
 public:
-    RunLoop(uv_loop_t*);
-    ~RunLoop();
+    enum class Type : uint8_t {
+        Default,
+        New,
+    };
 
-    static RunLoop* Get() {
-        return current.get();
-    }
+    enum class Event : uint8_t {
+        None      = 0,
+        Read      = 1,
+        Write     = 2,
+        ReadWrite = Read | Write,
+    };
 
-    static uv_loop_t* getLoop() {
-        return current.get()->get();
-    }
+    RunLoop(Type type = Type::Default);
+    ~RunLoop();
 
+    static RunLoop* Get();
+    static LOOP_HANDLE getLoopHandle();
```

Comment:
Yeah, I agree, I'm not too worried. I'll see if anything springs to mind after all the refactorings here shake out.